### PR TITLE
[Intl] Update the ICU data to 63.1

### PR DESCRIPTION
**Pull request by jakzal**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

http://site.icu-project.org/download/63

All tests are passing on the CI. I additionally run intl related tests (Form,Intl,Translation,Validator components) with ICU `63.1` on PHP `5.5`, `5.6`, `7.1`, `7.2`, `7.3-rc`.

Confirmed `@group intl-data` tests are passing with ICU `63.1` on PHP `5.5` and `7.2`.

### CODE REVIEW

**Comment by jakzal** [comment]
It's a work in progress since I still have to make all the tests pass.

File: src/Symfony/Component/Intl/Resources/bin/common.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ function run($command)
 
 function get_icu_version_from_genrb($genrb)
 {
-    exec($genrb.' --version 2>&1', $output, $status);
+    exec($genrb.' --version - 2>&1', $output, $status);
```
Comment (ro0NL) [inline src/Symfony/Component/Intl/Resources/bin/common.php]: thx for this :D

Comment (jakzal) [reply src/Symfony/Component/Intl/Resources/bin/common.php]: Thanks for finding a fix! :)

File: src/Symfony/Component/Form/Extension/Core/DataTransformer/NumberToLocalizedStringTransformer.php [hunk 2]
```diff
@@ -121,7 +121,7 @@ public function transform($value)
         }
 
         // Convert fixed spaces to normal ones
-        $value = str_replace("\xc2\xa0", ' ', $value);
+        $value = str_replace(array("\xc2\xa0", "\xe2\x80\xaf"), ' ', $value);
```
Comment (jakzal) [inline src/Symfony/Component/Form/Extension/Core/DataTransformer/NumberToLocalizedStringTransformer.php]: `e2 80 af` is a narrow no-break space. They seem to've started using it in the latest ICU.

File: src/Symfony/Component/Intl/NumberFormatter/NumberFormatter.php [hunk 3]
```diff
@@ -331,7 +331,7 @@ public function formatCurrency($value, $currency)
 
         $value = $this->formatNumber($value, $fractionDigits);
 
-        $ret = $symbol.$value;
+        $ret = $symbol.(mb_strlen($symbol, 'UTF-8') > 2 ? "\xc2\xa0" : '').$value;
```
Comment (jakzal) [inline src/Symfony/Component/Intl/NumberFormatter/NumberFormatter.php]: There's now a non-breaking space after the currency code (`CRC 100`), but not if the currency has a symbol (`£100`).

File: src/Symfony/Component/Intl/NumberFormatter/NumberFormatter.php [hunk 4]
```diff
@@ -513,17 +513,15 @@ public function parse($value, $type = self::TYPE_DOUBLE, &$position = 0)
             return false;
         }
 
-        $groupSep = $this->getAttribute(self::GROUPING_USED) ? ',' : '';
-
-        // Any string before the numeric value causes error in the parsing
-        if (preg_match("/^-?(?:\.\d++|([\d{$groupSep}]++)(?:\.\d*+)?)/", $value, $matches)) {
+        $groupingMatch = $this->getAttribute(self::GROUPING_USED) ? '|(?P<grouping>\d++(,{1}\d+)++(\.\d*+)?)' : '';
+        if (preg_match("/^-?(?:\.\d++{$groupingMatch}|\d++(\.\d*+)?)/", $value, $matches)) {
```
Comment (jakzal) [inline src/Symfony/Component/Intl/NumberFormatter/NumberFormatter.php]: I have added quite a few test cases to make sure no behaviour was broken by changes introduced in this code block: [AbstractNumberFormatterTest::testParse()]( https://github.com/symfony/symfony/pull/28977/files#diff-542534c8af70bcf20dd77bededfdb7c3R669)
Main change I noticed is that `-123,,456` didn't use to return a result, while now it parses as `-123`.

**Comment by jakzal** [comment]
All tests passing. PR ready.

**Comment by fabpot** [comment]
@jakzal Thanks for working on this. I'm wondering if we could add your Github's comment directly as comments in the code. They are valuable and should be kept close to the code IMHO. WDYT?

**Comment by jakzal** [comment]
Good point. Added comments.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jakzal.

**Pull request merged**